src/components/instances_page.rs

```diff
@@ -40,12 +41,32 @@ impl SimpleComponent for InstancesPage {
             #[name(stack)]
             gtk::Stack {
                 add_child = &gtk::ScrolledWindow {
-                    #[local_ref]
-                    instances -> gtk::Box {
+                    set_vexpand: true,
+                    set_hexpand: true,
+                    gtk::Box {
                         set_orientation: gtk::Orientation::Vertical,
-                        set_spacing: 5,
-                        set_vexpand: true,
-                    },
+                        set_spacing: 10,
+                        set_margin_all: 10,
+
+                        gtk::Box {
+                            set_spacing: 10,
+                            gtk::Entry {
+                                set_hexpand: true,
+                                set_tooltip_text: Some("Search"),
+                                set_buffer: &model.instances_search_buffer,
+                            },
+                            gtk::Button {
+                                set_label: "Filter",
+                                connect_clicked => InstancesPageInput::FetchInstances,
+                                }
+                            },
+                        #[local_ref]
+                        instances -> gtk::Box {
+                            set_orientation: gtk::Orientation::Vertical,
+                            set_spacing: 5,
+                            set_vexpand: true,
+                            },
+                        }
```

Comment:
nit: these two lines should be indented one tab less

Reply:
Fixed: bc468b7240848624007b3173fc5b3bd1d99a0b4a